### PR TITLE
chore(maintenance): migrate parser utility to biome

### LINKED ISSUE #2801: Maintenance: add biome to project (parser)
### Summary

With #2791 being merged, we can now migrate the `packages/parser` workspace to use Biome.

As part of the migration I will fix any issue/inconsistency that arises. I'll do my best to minimize changes and leave the logic untouched.

### Why is this needed?

So that we can gradually replace `eslint`, `prettier`, and all their dependencies with a single one, `@biomejs/biome`.

### Which area does this relate to?

Other

### Solution

_No response_

### Acknowledgment

- [X] This request meets [Powertools for AWS Lambda (TypeScript) Tenets](https://docs.powertools.aws.dev/lambda/typescript/latest/#tenets)
- [ ] Should this be considered in other Powertools for AWS Lambda languages? i.e. [Python](https://github.com/aws-powertools/powertools-lambda-python/), [Java](https://github.com/aws-powertools/powertools-lambda-java/), and [.NET](https://github.com/aws-powertools/powertools-lambda-dotnet/)

### Future readers

Please react with 👍 and your use case to help us understand customer demand.

### PR DESCRIPTION
**Pull request by daschaa**
### Changes

**Issue number: closes #2801** 

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Disclaimer**: We value your time and bandwidth. As such, any pull requests created on non-triaged issues might not be successful.

### CODE REVIEW

File: packages/parser/tests/unit/schema/s3.test.ts [hunk 1]
```diff
@@ -102,4 +104,21 @@ describe('S3 ', () => {
       parsed.userIdentity?.sessionContext?.attributes.mfaAuthenticated
     ).toEqual(false);
   });
+
+  it('should throw error when s3 event delete object does not contain etag or size', () => {
```
Comment (dreamorosi) [inline packages/parser/tests/unit/schema/s3.test.ts:108]: How come did we add a new test? Did the formatting change the logic?

**Comment by dreamorosi** [comment]
It seems that there is an extra unused import, please check the SonarCloud finding before we merge

File: packages/parser/tests/unit/schema/s3.test.ts (hunk 1 above)
Comment (dreamorosi) [reply packages/parser/tests/unit/schema/s3.test.ts:108]: I just saw your message on Discord! - let me take a look

Comment (dreamorosi) [reply packages/parser/tests/unit/schema/s3.test.ts:108]: Ok, so, for everyone reading this later: it seems that we might have an inconsistency in the `S3RecordSchema` schema (to be confirmed).

We know from previous discussions ([here](https://github.com/aws-powertools/powertools-lambda-python/issues/4156#issuecomment-2097901803) & [here](https://github.com/aws-powertools/powertools-lambda-typescript/issues/2410#issue-2256226879)) that:
- `etag` (EventBridge->S3) and `eTag` (S3->Lambda/SNS/SQS) are omitted for Object Deletion events
- `size` is omitted for Object Deletion events

The `.refine()` method in the schema however seems to be asserting the opposite.

Looping in @am29d who knows this area of the codebase better than me.

**Comment by daschaa** [comment]
@dreamorosi regarding the sonar issue: I thought I already removed the unused import. I also don't see the change anymore in the changed files. Is it possible to re-trigger the Sonar scan?

File: packages/parser/tests/unit/schema/s3.test.ts (hunk 1 above)
Comment (daschaa) [reply packages/parser/tests/unit/schema/s3.test.ts]: As discussed, we agreed to remove the `.refine()` method from the schema.

**Pull request merged**